github/messages_test.go

```diff
@@ -79,3 +81,110 @@ func TestValidatePayload(t *testing.T) {
 		}
 	}
 }
+
+func TestGetWebHookEvent(t *testing.T) {
+	tests := []struct {
+		payload     interface{}
+		messageType string
+	}{
+		{
+			payload:     &CommitCommentEvent{},
+			messageType: "commit_comment",
+		},
+		{
+			payload:     &CreateEvent{},
+			messageType: "create",
+		},
+		{
+			payload:     &DeleteEvent{},
+			messageType: "delete",
+		},
+		{
+			payload:     &DeploymentEvent{},
+			messageType: "deployment",
+		},
+
+		{
+			payload:     &DeploymentStatusEvent{},
+			messageType: "deployment_status",
+		},
+		{
+			payload:     &ForkEvent{},
+			messageType: "fork",
+		},
+		{
+			payload:     &GollumEvent{},
+			messageType: "gollum",
+		},
+		{
+			payload:     &IssueCommentEvent{},
+			messageType: "issue_comment",
+		},
+		{
+			payload:     &IssuesEvent{},
+			messageType: "issues",
+		},
+		{
+			payload:     &MemberEvent{},
+			messageType: "member",
+		},
+		{
+			payload:     &MembershipEvent{},
+			messageType: "membership",
+		},
+		{
+			payload:     &PageBuildEvent{},
+			messageType: "page_build",
+		},
+		{
+			payload:     &PublicEvent{},
+			messageType: "public",
+		},
+		{
+			payload:     &PullRequestEvent{},
+			messageType: "pull_request",
+		},
+		{
+			payload:     &PullRequestReviewCommentEvent{},
+			messageType: "pull_request_review_comment",
+		},
+		{
+			payload:     &PushEvent{},
+			messageType: "push",
+		},
+		{
+			payload:     &ReleaseEvent{},
+			messageType: "release",
+		},
+		{
+			payload:     &RepositoryEvent{},
+			messageType: "repository",
+		},
+		{
+			payload:     &StatusEvent{},
+			messageType: "status",
+		},
+		{
+			payload:     &TeamAddEvent{},
+			messageType: "team_add",
+		},
+		{
+			payload:     &WatchEvent{},
+			messageType: "watch",
+		},
+	}
+
+	for _, test := range tests {
+		p, err := json.Marshal(test.payload)
+		if err != nil {
+			t.Fatalf("Marshal(%#v): %v", test.payload, err)
+		}
+		got, err := ParseWebHook(test.messageType, p)
+		if err != nil {
+			t.Fatalf("getWebHookEvent: %v", err)
+		}
+		if !reflect.DeepEqual(test.payload, got) {
```

Comment:
The common style is to compare got with want, in that order. Not the other way around.

```
github $ cat *.go | grep "got, want" | wc -l
     152
github $ cat *.go | grep "want, got" | wc -l
       0
```

It might be more readable as:

```Go
if want := test.payload; !reflect.DeepEqual(got, want) {
```